File: docs/examples/teaching/10-local-logic/passive/recipe.py
from gramlot.page import WebPage


class Page(WebPage):
    def main(self, root):
        root.numberTextBox(value="^quantity", lbl="Quantity", live=True)
        root.numberTextBox(value="^price", lbl="Price", live=True)
        root.dataFormula("total", "quantity * price", quantity="^quantity", price="=price")
        root.p("^total")

File: docs/examples/teaching/10-local-logic/passive/recipe.js
export function build(root) {
    root.numberTextBox({value: '^quantity', lbl: 'Quantity', live: true});
    root.numberTextBox({value: '^price', lbl: 'Price', live: true});
    root.dataFormula({destination: 'total', formula: 'quantity * price', quantity: '^quantity', price: '=price'});
    root.p('^total');
}
